### DISCUSSION TITLE
Does `mpv` have to depend on `youtube-dl`?

### DISCUSSION THREAD

**Discussion by triallax** [Homebrew/discussions#2220] [Everyday usage]
I think it shouldn't, for the following reasons:
1. MPV doesn't require youtube-dl to run, so it's one more unnecessary formula for people who don't use it
2. Some people might want to use youtube-dl forks (in fact, I opened this discussion because I'm trying to move to [`yt-dlp`](https://github.com/yt-dlp/yt-dlp) instead)

**Comment by SMillerDev** [comment]
What does mpv define in their documentation?

**Reply by triallax** [reply]
Sorry, I didn't understand. Can you clarify?

**Reply by triallax** [reply]
Actually, also of interest here is that MPV now searches for `yt-dlp` by default (https://github.com/mpv-player/mpv/pull/9209), and if it's not found, _then_ it looks for `youtube-dl`, though both are still not required. An option here is to depend on `yt-dlp` instead of `youtube-dl`, but I say to depend on neither and let the users handle this.

**Reply by SMillerDev** [reply]
What happens if it can't find either one?

**Reply by triallax** [reply]
If we're talking about a URL, if it can't play it without either (if it's a YouTube link for example), it will error. If it's a file or the URL leads directly to a video then it just works.
